x-pack/plugins/security_solution/server/lib/telemetry/endpoint.ts

```diff
@@ -0,0 +1,131 @@
+export interface PoliciesTelemetry {
+  malware: {
+    success: number;
+    warning: number;
+    failure: number;
+  };
+}
+
+export interface AgentMetadataTelemetry {
+  total_installed: number;
+  active_within_last_24_hours: number;
+  os: AgentOSMetadataTelemetry[];
+  policies?: PoliciesTelemetry; // TODO: make required when able to enable policy information
+}
+
+export interface AgentLocalMetadata extends AgentMetadata {
+  elastic: {
+    agent: {
+      id: string;
+    };
+  };
+  host: {
+    id: string;
+  };
+  os: {
+    name: string;
+    platform: string;
+    version: string;
+    full: string;
+  };
+}
+
+export const getDefaultEndpointTelemetry = (): AgentMetadataTelemetry => ({
+  total_installed: 0,
+  active_within_last_24_hours: 0,
+  os: [],
+});
+
+export const getEndpointTelemetryFromFleet = async (
+  savedObjectsClient: SavedObjectsClientContract
+) => {
+  // Retrieve every agent that references the endpoint as an installed package. It will not be listed if it was never installed
+  const { saved_objects: endpointAgents } = await getFleetSavedObjectsMetadata(savedObjectsClient);
+  const endpointTelemetry = getDefaultEndpointTelemetry();
+  // If there are no installed endpoints return the default telemetry object
+  if (!endpointAgents || endpointAgents.length < 1) return endpointTelemetry;
+
+  const uniqueHostIds: Set<string> = new Set();
+  const uniqueAgentIds: Set<string> = new Set();
+
+  // All agents who have the endpoint registered as an installed package.
+  endpointTelemetry.total_installed = endpointAgents.length;
+
+  const aDayAgo = new Date();
+  aDayAgo.setDate(aDayAgo.getDate() - 1);
+
+  const osTracker: Record<string, AgentOSMetadataTelemetry> = {};
+
+  const endpointMetadataTelemetry = endpointAgents.reduce(
+    (metadataTelemetry, { attributes: metadataAttributes }) => {
+      const { last_checkin: lastCheckin, local_metadata: localMetadata } = metadataAttributes;
+      const { host, os, elastic } = localMetadata as AgentLocalMetadata;
+
+      if (lastCheckin && new Date(lastCheckin) > aDayAgo) {
+        // Get agents that have checked in within the last 24 hours to later see if their endpoints are running
+        uniqueAgentIds.add(elastic.agent.id);
+      }
+      if (host && uniqueHostIds.has(host.id)) {
+        return metadataTelemetry;
+      } else {
+        uniqueHostIds.add(host.id);
+        const { name: osName, version: osVersion, platform: osPlatform, full: osFullName } = os;
+        if (osFullName && osVersion) {
+          if (osTracker[osName]) osTracker[osName].count += 1;
+          else {
+            osTracker[osName] = {
+              full_name: osFullName,
+              platform: osPlatform,
+              version: osVersion,
+              count: 1,
+            };
+          }
+        }
+        return metadataTelemetry;
+      }
+    },
+    endpointTelemetry
+  );
+
+  endpointMetadataTelemetry.os = Object.values(osTracker);
+
+  // Handle Last 24 Hours
```

Comment:
Currently, the best way to know for certain that an endpoint is actively running is to check the latest status the agent has for the endpoint from it's events list